RELEASE/scripts/sl_ascend/sl_cooking.ash

```diff
@@ -1648,111 +1784,146 @@ boolean loadConsumables(string _type, item[int] item_backmap, int[int] cafe_back
 				howmany = min(howmany, max(0, creatable_amount(it) - sl_reserveCraftAmount(it)));
 				craftables[it] = howmany;
 			}
+			if (storage_amount(it) > 0)
+			{
+				// We could have more stringent requirements here...
+				howmany = min(howmany, min(pulls_remaining()/2 - 2, storage_amount(it)));
+				pullables[it] = howmany;
+			}
 		}
 	}
 
-	void add(item it, boolean crafting, int howmany)
+	boolean wantBorisPie = false;
+	boolean wantJarlsbergPie = false;
+	boolean wantPetePie = false;
+
+	if ((item_amount($item[Boris's key]) == 0) && (item_amount($item[Jarlsberg's key]) == 0) && (item_amount($item[Sneaky Pete's key]) == 0) && towerKeyCount() < 3)
+	{
+		// I refuse to handle the hard case where someone has pulled and eaten a key
```

Comment:
is it really that hard? Wouldn't it be easy to just take this whole block and replace it with

```
if(towerKeyCount() < 3)
{
  if(item_amount($item[Boris's key]) == 0)
    wantBorisPie = true;
  if(item_amount($item[Jarlsberg's key]) == 0)
    wantJarlsbergPie = true;
  if(item_amount($item[Sneaky Pete's key]) == 0)
    wantPetePie = true;
}
```

Or am I missing something there?

Reply:
We don't cash in our tokens for keys until we're at the tower door.

Reply:
So like, how about

``
if(towerKeyCount() < 3)
{
  if(item_amount($item[Boris's key]) == 0 && item_amount($item[phat loot token]) < 3)
    wantBorisPie = true;
  if(item_amount($item[Jarlsberg's key]) == 0 && item_amount($item[phat loot token]) < 2)
    wantJarlsbergPie = true;
  if(item_amount($item[Sneaky Pete's key]) == 0 && item_amount($item[phat loot token]) < 1)
    wantPetePie = true;
}
``